### PR TITLE
99_custom_models_gui_support

### LINKED ISSUE #99: Improve support for custom models
#98 will introduce fine-tuned models. (They are still preliminary and will be updated + specific models for electron microscopy will be added).
We should improve the support for it by:
- Enable selecting them from the annotator GUI
- Keeping track of which model was used for producing embedding file (extending what we currently have for the data signature)
    - We should also check for the tiling there
- Possibly making the LM models the defaults

### PR DESCRIPTION
**Pull request by paulhfu**
Fixes #99
add available model types into gui from dict in utils

### CODE REVIEW

**Review by constantinpape — COMMENTED** [review]
Thanks! This looks good, I just think there is one possible improvement: make `model_type` an attribute of `predictor` instead of introducing it as a new argument to `precompute_image_embeddings`. This way we avoid that these two get out of sync and reduce the complexity of `precomute_image_embeddings`.

File: micro_sam/util.py [hunk 1]
```diff
@@ -372,6 +377,7 @@ def precompute_image_embeddings(
         ndim: The dimensionality of the data. If not given will be deduced from the input data.
         tile_shape: Shape of tiles for tiled prediction. By default prediction is run without tiling.
         halo: Overlap of the tiles for tiled prediction.
+        model_type: Name of the model. Should be in util.get_model_names().
```
[the remark this answers is not in the document]
Comment (paulhfu) [reply micro_sam/util.py:380]: yes, makes sense.

**Comment by paulhfu** [comment]
I noticed, that this warning dialog (wrong embeddings file) is not working anymore.
My suspicion is, that this is due to the destruction of the QT event loop [here](https://github.com/computational-cell-analytics/micro-sam/blob/31a8a3d8204d812c0580c75f95238e66172c1563/micro_sam/sam_annotator/annotator.py#L466)

I wanted to fix this issue also in this PR.

**Comment by paulhfu** [comment]
This can be reviewed now.

File: micro_sam/sam_annotator/gui_utils.py [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import os
+import magicgui
 from shutil import rmtree
 from typing import Union
 
-from PyQt5 import QtCore, QtWidgets
```
Comment (constantinpape) [inline micro_sam/sam_annotator/gui_utils.py:5]: The import 
```
from PyQt5 import QtWidgets
```
is still needed, see e.g. line 50.

**Review by constantinpape — COMMENTED** [review]
The `QTWidgets` import is missing. I also checked locally and the `Create a new file` option is behaving a bit weird after adding the import back.
I will share a video to show.

**Comment by constantinpape** [comment]
Here's how the `Create new file` option is behaving:
the `Open` button needs to pressed multiple times (but then it works.)

https://github.com/computational-cell-analytics/micro-sam/assets/4263537/8a4c0309-4257-49d9-9efd-4157bcb3fa93

**Comment by paulhfu** [comment]
Thanks for testing @constantinpape. I fixed the reported issues.

**Review by constantinpape — APPROVED** [review]
I checked again and all is working now.
Thanks!

**Pull request merged**